nether/portal.lua

```diff
@@ -41,6 +41,47 @@ local function save_nether_players()
 	io.close(f)
 end
 
+-- Nether aware mods will need to know if a player is in the nether.
+function nether.is_player_in_nether(player)
+	local pos = player:get_pos()
+	return (pos.y < nether.start) and (pos.y >= nether.bottom)
+end
+
+-- For testing nether trap state tracking.
+function nether.is_player_trapped_in_nether(player)
+	return players_trapped_in_nether[player:get_player_name()]
+end
+
+-- Nether aware mods may have other means of moving players between the Nether
+-- and Overworld, and if so, they should tell us about it so we can keep track
+-- of the player state.
+function nether.external_nether_teleport(player, pos)
+	if not nether.trap_players then
+		player:set_pos(pos)
+		return
+	end
+	local destination_in_nether = (pos.y < nether.start) and (pos.y >= nether.bottom)
+	update_background(player, destination_in_nether)
+	local pname = player:get_player_name()
+	players_trapped_in_nether[pname] = destination_in_nether or nil
+	player:set_pos(pos)
+end
+
+-- Has the player dug their way out of the nether?
+-- Has nether.trap_players been disabled?
+function nether.registry_update(player)
+	local pos = player:get_pos()
+	local in_nether = (pos.y < nether.start) and (pos.y >= nether.bottom)
+	local pname = player:get_player_name()
+	if nether.trap_players then
+		players_trapped_in_nether[pname] = in_nether
```

Comment:
`save_nether_players` saves all keys in `players_trapped_in_nether` to the nether_players file even if the value is false, so `players_trapped_in_nether[pname]` should either be unset (nil) or true.
```lua
		players_trapped_in_nether[pname] = in_nether or nil
```

Alternatively, the `save_nether_players` function can be changed to check if the value evaluates to a truth value before saving the player name.

Reply:
Fixed